vminfo_parser/analyzer.py

```diff
@@ -264,44 +362,62 @@ def handle_disk_space(
         else:
-            range_counts_by_environment = (
-                dataFrame[dataFrame[envHeading] == environment_filter]
-                .groupby(["Disk Space Range", envHeading])
-                .size()
-                .unstack(fill_value=0)
+            column_padding, sorted_range_counts_by_environment = self.sort_by_disk_space_range(
+                dataFrame,
+                drop_columns=["second_number"],
+                os_breakdown=False,
+                environment_filter=environment_filter,
             )
 
-        # create an integer of the large end of range for sorting by size of range
-        range_counts_by_environment["second_number"] = (
-            range_counts_by_environment.index.str.split("-").str[1].str.split().str[0].astype(int)
-        )
-        sorted_range_counts_by_environment = range_counts_by_environment.sort_values(by="second_number", ascending=True)
-        sorted_range_counts_by_environment.drop("second_number", axis=1, inplace=True)
+        if env_keywords and environment_filter != "all":
+            if granular_disk_space_by_os:
+                col_widths = self.cli_output.set_column_width(
+                    sorted_range_counts_by_environment,
+                    index_column_padding=0,
+                    remaining_column_padding=19,
+                    index_column_name="OS Version Number",
+                )
+            else:
+                col_widths = self.cli_output.set_column_width(
+                    sorted_range_counts_by_environment,
+                    index_column_padding=22,
+                    remaining_column_padding=10,
+                    index_column_name="Environment",
+                )
+        else:
+            col_widths = self.cli_output.set_column_width(
+                sorted_range_counts_by_environment, index_column_padding=17, remaining_column_padding=17
+            )
 
-        # Apply the conversion to the index
-        sorted_range_counts_by_environment.index = sorted_range_counts_by_environment.index.map(self.convert_to_tb)
+        formatted_rows = []
 
+        if environment_filter == "all":
+            formatted_rows.append("Disk Space Range".ljust(20) + "Count".ljust(col_widths["Count"]))
+            column_padding["justification"] = 19
+
+        formatted_df_str = self.cli_output.format_rows(
+            sorted_range_counts_by_environment, formatted_rows, column_padding["justification"], col_widths
+        )
         # Now call the print_formatted_disk_space method
         self.cli_output.print_formatted_disk_space(
-            sorted_range_counts_by_environment,
-            environment_filter,
-            env_keywords,
+            col_widths,
+            formatted_df_str,
             os_filter=os_filter,
+            index_heading_justification=column_padding["index_heading_justification"],
+            other_headings_justification=column_padding["other_headings_justification"],
         )
```

Comment:
Why is format_rows being called by analyzer, and brought back into analyzer if its only sent back to cli_output?

Reply:
I was trying to break apart  the print_formatted_disk_space. Initially it had all of this logic inside of it, including set_column_width()

Flake8 was tripping over the complexity with both of these tasks in the print_formatted_disk_space() so I broken them up.

Reply:
I agree with splitting them up,  but I think that the logic can and should be called from within clioutput, unless the output is needed in analyzer for something else

Reply:
If format_rows is only useful to cli_output,  then analyzer shouldn't know or care about it,  ultimately, the goal is analyzer doesn't care what the output is,  its just crunching data.